cmd/syncthing/monitor.go

```diff
@@ -181,6 +186,43 @@ func copyStderr(stderr io.ReadCloser, dst io.Writer) {
 					panicFd.WriteString(line)
 				}
 				stdoutMut.Unlock()
+
+				if optPanicReporting {
+					// Prepare query for collecting error log
+					type Report struct {
+						Error       string
+						Version     string
+						LongVersion string
+						Os          string
+						Arch        string
+					}
+					report := &Report{Error: line[:len(line)-1],
+						Version:     Version,
+						LongVersion: LongVersion,
+						Os:          runtime.GOOS,
+						Arch:        runtime.GOARCH}
+					b, err := json.Marshal(report)
+					if err == nil {
```

Comment:
I am afraid this will change the behaviour. There is an instruction after 'if' but in the 'for' that we do not want to ommit.